docs/tutorials/todo-manager/README.mdx

```diff
@@ -0,0 +1,1097 @@
+
+    def get_todo_by_id(self, id: str) -> Optional[Todo]:
+        return next((todo for todo in self.todos if todo.id == id), None)
+
+    def create_todo(self, content: str, owner_id: str) -> Todo:
+        todo = Todo(
+            id=self._gen_id(),
+            content=content,
+            owner_id=owner_id,
+            created_at=datetime.now().isoformat()
+        )
+        self.todos.append(todo)
+        return todo
+
+    def delete_todo(self, id: str) -> Optional[Todo]:
+        for i, todo in enumerate(self.todos):
+            if todo.id == id:
+                return self.todos.pop(i)
+        return None
+
+    def _gen_id(self) -> str:
+        """Generate a random 8-character string for todo ID."""
+        return ''.join(random.choices(string.ascii_lowercase + string.digits, k=8))
+```
+
+
+</TabItem>
+<TabItem value="node" label="Node.js">
+
+```ts
+// todo-service.ts
+
+type Todo = {
+  id: string;
+  content: string;
+  ownerId: string;
+  createdAt: string;
+};
+
+/**
+ * A simple Todo service for demonstration purposes.
+ * Use an in-memory array to store todos
+ */
+export class TodoService {
+  private readonly todos: Todo[] = [];
+
+  getAllTodos(ownerId?: string): Todo[] {
+    if (ownerId) {
+      return this.todos.filter((todo) => todo.ownerId === ownerId);
+    }
+    return this.todos;
+  }
+
+  getTodoById(id: string): Todo | undefined {
+    return this.todos.find((todo) => todo.id === id);
+  }
+
+  createTodo({ content, ownerId }: { content: string; ownerId: string }): Todo {
+    const todo: Todo = {
+      id: this.genId(),
+      content,
+      ownerId,
+      createdAt: new Date().toISOString(),
+    };
+
+    // eslint-disable-next-line @silverhand/fp/no-mutating-methods
+    this.todos.push(todo);
+    return todo;
+  }
+
+  deleteTodo(id: string): Todo | undefined {
+    const index = this.todos.findIndex((todo) => todo.id === id);
+
+    if (index === -1) {
+      return undefined;
+    }
+
+    // eslint-disable-next-line @silverhand/fp/no-mutating-methods
+    const [deleted] = this.todos.splice(index, 1);
+    return deleted;
+  }
+
+  private genId(): string {
+    return Math.random().toString(36).slice(2, 10);
+  }
+}
+```
+</TabItem>
+</Tabs>
+
+then in the tools layer, we'll determine whether operations are allowed based on the user's scopes:
+
+<Tabs groupId="sdk">
+<TabItem value="python" label="Python">
+
+```python
+# todo-manager.py
+
+from typing import Any, Optional
+from mcpauth.errors import MCPAuthBearerAuthError
+
+def assert_user_id(auth_info: Optional[dict]) -> str:
+    """Extract and validate user ID from auth info."""
+    subject = auth_info.get('subject') if auth_info else None
+    if not subject:
+        raise ValueError('Invalid auth info')
+    return subject
+
+def has_required_scopes(user_scopes: list[str], required_scopes: list[str]) -> bool:
+    """Check if user has all required scopes."""
+    return all(scope in user_scopes for scope in required_scopes)
```

Comment:
non-blocking: it'll be better if we can have something like this in mcp-auth SDKs